met-api/src/met_api/services/engagement_service.py

```diff
@@ -35,6 +42,14 @@ def get_all_engagements(user_id):
 
         return engagements
 
+    @staticmethod
+    def close_engagements_due():
+        """Close published engagements that are due for a closeout."""
+        engagements = Engagement.close_engagements_due()
+        print('Engagements to close: ', engagements)
+        results = [EngagementService._send_closeout_emails(engagement) for engagement in engagements]
```

Comment:
what happens if sending closeout email to engagement 3 of 5 per se failed and raised exception. What happens regarding sending emails for 4 and 5

Reply:
Maybe we can research sending the emails asynchronously and not let failure of one affect the others. When all the emails return (successes and fails) we can say failed to send emails for engagements: 3 and succeeded for: 1 2 4 5

Reply:
@saravanpa-aot possible, I know this is doable in Java

Reply:
hm.. we will move to an async model later ;we need to implement a messaging component ;like a queue..once we have it , it will be easier..

Reply:
I will create a ticket for this..Thanks for reminding @jadmsaadaot